### PR TITLE
Addon-docs: Props / Description for web-components

### PR DESCRIPTION
**Pull request by daKmoR**
Issue:

## What I did

- Use MetaData from `custom-elements.json`
- Show props table in docs mode for web-components
- Show description
- Added to example

## How to test

open kitchen sink docs mode

## Still open

- [x] Document how to setup/use
- [x] Types are not shown?

=> if approach looks good I will add documentation for it 👍 

as now.sh is currently having some trouble here is a screenshot how it will look like
Screenshot 2019-10-27 at 14 43 03

### CODE REVIEW

File: examples/web-components-kitchen-sink/.storybook/config.js [hunk 1]
```diff
@@ -3,6 +3,11 @@
 import { configure, addParameters, addDecorator } from '@storybook/web-components';
 import { withA11y } from '@storybook/addon-a11y';
 
+import customElements from '../custom-elements.json';
+
+// eslint-disable-next-line no-underscore-dangle
+window.__STORYBOOK_CUSTOM_ELEMENTS__ = customElements;
```
Comment (shilman) [inline examples/web-components-kitchen-sink/.storybook/config.js]: how about:

```js
import { setCustomElements } from '@storybook/addon-docs/web-components';

...

setCustomElements(customElements);
```

Comment (daKmoR) [reply examples/web-components-kitchen-sink/.storybook/config.js]: great idea 👍

Comment (daKmoR) [reply examples/web-components-kitchen-sink/.storybook/config.js]: done and pushed 💪

File: addons/docs/src/frameworks/web-components/config.js [hunk 2]
```diff
@@ -24,7 +24,7 @@ addParameters({
     page: DocsPage,
     extractProps: tagName => {
       const customElements = getCustomElements();
-      if (customElements) {
+      if (typeof tagName === 'string' && customElements && customElements.tags) {
```
Comment (shilman) [inline addons/docs/src/frameworks/web-components/config.js]: Maybe `checkValidComponent(tagName)` that throws if it's not a string?

File: addons/docs/README.md [hunk 3]
```diff
@@ -75,16 +75,16 @@ For more information on `MDX`, see the [`MDX` reference](./docs/mdx.md).
...
+| MDX stories       |   +   |  +  |    +    |  +   |                 +                  |   +    |    +    |   +   |    +    |  +   |   +   |   +    |
+| CSF stories       |   +   |  +  |    +    |  +   |                 +                  |   +    |    +    |   +   |    +    |  +   |   +   |   +    |
+| StoriesOf stories |   +   |  +  |    +    |  +   |                 +                  |   +    |    +    |   +   |    +    |  +   |   +   |   +    |
+| Source            |   +   |  +  |    +    |  +   |                 +                  |   +    |    +    |   +   |    +    |  +   |   +   |   +    |
+| Notes / Info      |   +   |  +  |    +    |  +   |                 +                  |   +    |    +    |   +   |    +    |  +   |   +   |   +    |
+| Props table       |   +   |  +  |    #    |      |                 +                  |        |         |       |         |      |       |        |
+| Docgen            |   +   |  +  |    #    |      |                                    |        |         |       |         |      |       |        |
```
Comment (shilman) [inline addons/docs/README.md]: Docgen => Description, add +

File: addons/docs/src/frameworks/web-components/config.js (hunk 2 above)
Comment (daKmoR) [reply addons/docs/src/frameworks/web-components/config.js]: if nothing is provided we should not throw...

but if something is provided which is different then a string then we should 👍 
added 💪

File: addons/docs/README.md (hunk 3 above)
Comment (daKmoR) [reply addons/docs/README.md]: nice 🤗 change + pushed

**Review by shilman — APPROVED** [review]
SUPERB!!! 🏆

**Comment by alazurenko** [comment]
@daKmoR Thank you for this contribution! 
I was looking for solution to documentation props of web-components in storybook/docs!

I have a question regarding the way how this json is generated. Is it possible to automate this process of generation of properties?

**Comment by alazurenko** [comment]
@daKmoR 
Maybe it was just me not paying much attention 🙂 Not it is way more clearer. 

Thank you for help!

**Comment by beth-panx** [comment]
Hello @daKmoR quick question for you - should properties be called 'props' for web components? 🤓

**Comment by daKmoR** [comment]
hmm properties would probably be better 🤔

care to create a PR and update it? 🤗

**Comment by beth-panx** [comment]
> hmm properties would probably be better 🤔
> 
> care to create a PR and update it? 🤗

Sure thing. PR is created [here](https://github.com/storybookjs/storybook/pull/9362).

**Pull request merged**